src/GoRestUsersApi.js

```diff
@@ -0,0 +1,72 @@
+const request = require("request-promise");
+
+const GOREST_HOST = "https://gorest.co.in";
+const publicPrefix = '/public/v1';
+const GOREST_APIKEY = "cff60f4830111a45ffa352fbc2aae7cdc4b27ca0911a795372ee9cd2ca6f09bf";
+
+class UsersApi {
+    constructor(host = GOREST_HOST) {
+      this.host = host;
+      this.request = request.defaults({
+        headers: {
+            "Content-Type": "application/json", 
+            "Accept":"application/json"                                  
+        },
+        json: true        
+      })
+    }
+  
+  authenticate(apikey = GOREST_APIKEY) {
+    this.request = request.defaults({
+      headers: {
+          "Content-Type": "application/json", 
+          "Accept":"application/json",  
+          "Authorization":`Bearer ${apikey}`                     
+      },
+      json: true        
+    })
+  }
+
+  deleteUser(userId) {
+    const path = `/${publicPrefix}/users/${userId}`;
+
+    return this.request.delete({
+        url: `${this.host}${path}`       
+      })
+  }
+
+  updateUser(userId) {
```

Comment:
When you update something, you would need to know two things: 
- what you are updating (resource ID)
- what you want to update it with (body)

You are missing the latter.